js/glowingbear.js

```diff
@@ -761,26 +763,30 @@ weechat.controller('WeechatCtrl', ['$rootScope', '$scope', '$store', '$timeout',
         $scope.updateShowNicklist();
     });
     $scope.showNicklist = false;
-    // Utility function that template can use to check if nicklist should
-    // be displayed for current buffer or not
-    // is called on buffer switch and certain swipe actions
+    // Utility function that template can use to check if nicklist should be
+    // displayed for current buffer or not is called on buffer switch and
+    // certain swipe actions.  Sets $scope.showNicklist accordingly and returns
+    // whether the buffer even has a nicklist to show.
     $scope.updateShowNicklist = function() {
-        $scope.showNicklist = (function() {
+        // returns array of booleans: [show nicklist, buffer has nicklist]
```

Comment:
this should probably update $scope.showNicklist and return the value directly, without wrapping in another function and doing the stupid wrapping dance.